jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/ProxyProvider.java

```diff
@@ -18,33 +18,58 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
 
 /** Propagates proxy configuration from Maven settings to system properties. */
 class ProxyProvider {
 
   private static final ImmutableList<String> PROXY_PROPERTIES =
       ImmutableList.of("proxyHost", "proxyPort", "proxyUser", "proxyPassword");
 
+  private static final ImmutableList<String> PROXY_PROTOCOLS = ImmutableList.of("https", "http");
   /**
    * Initializes proxy settings based on Maven settings.
    *
    * @param settings Maven settings
    */
-  static void init(DecryptedMavenSettings settings) {
-    configureProxy(settings, "https");
-    configureProxy(settings, "http");
-  }
+  static void populateSystemProxyProperties(Settings settings, SettingsDecrypter decrypter)
+      throws MojoExecutionException {
+    List<Proxy> proxies =
+        settings
+            .getProxies()
+            .stream()
+            .filter(Proxy::isActive)
+            .filter(proxy -> PROXY_PROTOCOLS.contains(proxy.getProtocol()))
+            .collect(Collectors.toList());
 
-  private static void configureProxy(DecryptedMavenSettings settings, String protocol) {
-    settings
-        .getProxies()
-        .stream()
-        .filter(Proxy::isActive)
-        .filter(proxy -> protocol.equals(proxy.getProtocol()))
-        .findFirst()
-        .ifPresent(ProxyProvider::setProxyProperties);
+    System.out.println(proxies.size());
+    if (proxies.size() == 0) {
+      return;
+    }
+
+    SettingsDecryptionRequest request = new DefaultSettingsDecryptionRequest().setProxies(proxies);
+    System.out.println("Decrypter" + decrypter);
+    SettingsDecryptionResult result = decrypter.decrypt(request);
+
+    for (SettingsProblem problem : result.getProblems()) {
+      if (problem.getSeverity() == SettingsProblem.Severity.ERROR
+          || problem.getSeverity() == SettingsProblem.Severity.FATAL) {
+        throw new MojoExecutionException(
+            "Unable to decrypt proxy info from settings.xml: " + problem);
+      }
+    }
+
+    result.getProxies().forEach(ProxyProvider::setProxyProperties);
```

Comment:
This may not be right? What we've been doing is to call this twice on 1) the first active HTTP; and 2) the first active HTTPS. Looks like this code is calling this for all active proxies.

Reply:
this should be fixed now.